### ISSUE TITLE
documentLink should contain a target range

### ISSUE THREAD

**Issue by kjeremy** [microsoft/language-server-protocol#1173]
`DocumentLink` supports a `target` which is a `DocumentUri` and that can point inside a workspace. It should also provide an optional target range or position.

**Comment by prabirshrestha** [comment]
Can you provide more context?

**Comment by dbaeumer** [comment]
Would the range usually be encoded in the URI itself and depend on the target system (e.g. like `#L7` in GitHub https://github.com/microsoft/language-server-protocol/blob/master/License.txt#L7). So a server wouldn't be able to fill in the range in a general matter.

**Comment by kjeremy** [comment]
That's true for external documents. In rust we have intra-doc links [rustdoc](https://doc.rust-lang.org/rustdoc/linking-to-items-by-name.html) in markdown and I was hoping to resolve them using `documentLink` instead of rewriting the markdown returned from the server.

**Comment by dbaeumer** [comment]
Wouldn't this be better handled as a normal reference / definition. Then you would be able to Ctrl+Click and hover over it.

**Comment by dbaeumer** [comment]
@kjeremy any additional info?

**Comment by kjeremy** [comment]
I do not remember what my use case was... I think it was clicking on a documentLink in documentation that would bring me to a different spot in the file.

Example in the below code `[Bar]` would be a `documentLink` that pointed to the `Bar` in `pub struct Bar`.

```rust
/// This struct is not [Bar]
pub struct Foo1;

pub struct Bar;
```

**Comment by dbaeumer** [comment]
Hmm, why is this a document link and not a GoTo definition gesture?

**Comment by dbaeumer** [comment]
I will close the issue since I think that using definition / declaration / reference here. 

@kjeremy please ping if you still thing there is something to be done in the specifiction.
